File: src/state_handlers/ask_tutorial.js
/**
 * @fileOverview
 * @author Peter Mitrano- mitranopeter@gmail.com
 */

'use strict';

var Core = require('../core');
var Alexa = require('../alexa');

/**
 * Represents the reponses when we've just asked if the user wants to follow
 * a tutorial on how to use the app.
 */
module.exports = Alexa.CreateStateHandler(Core.states.ASK_TUTORIAL, {
  'AMAZON.YesIntent': function() {
    this.handler.state = Core.states.INITIAL_STATE;
    // don't end session here. It's convenient so that we don't
    // have say "ask my cookbook" again
    this.emit(":ask", "I am capable of finding recipes and walking you through making them. Try asking how to make pancakes");
    this.emit(":saveState", true);
  },
  'AMAZON.NoIntent': function() {
    this.handler.state = Core.states.PROMPT_FOR_START;
    this.emit(":ask", "Are you ready to start making something? You can say yes, or ask me something else.");
  },
  'Unhandled': function() {
    this.emit(":ask", "I'm confused. Do you want to start with a tutorial? Try saying yes or no.");
  }
});

File: my_cookbook/skill/handlers/ask_tutorial.py
from my_cookbook.util import core
from my_cookbook.util import responder


class AskTutorialHandler:
    def AMAZON_YesIntent(self, handlers, persistant_attributes, attributes, slots):
        return responder.tell(
            "I am capable of finding recipes and walking you through making them. Try asking how to make pancakes")

    def AMAZON_NoIntent(self, handlers, persistant_attributes, attributes, slots):
        attributes[core.STATE_KEY] = core.States.PROMPT_FOR_START
        return responder.ask(
            "Are you ready to start making something? You can say yes, or ask me something else.",
            None, attributes)

    def Unhandled(self, handlers, persistant_attributes, attributes, slots):
        return responder.ask(
            "I'm confused. Do you want to start with a tutorial? Try saying yes or no.", None,
            attributes)


handler = AskTutorialHandler()
state = core.States.ASK_TUTORIAL
